.circleci/config.yml

```diff
@@ -190,57 +191,90 @@ jobs:
           name: Setup python libraries
           command: |
               pip3 install requests pyyaml
+      - run:
+          name: Install dependencies
+          command: |
+              sudo apt-get update && \
+              sudo apt-get install jq unzip
       - run:
           name: Setup e2e-environment
           command: |
               source $PORTAL_SOURCE_DIR/env/custom.sh || true && \
-              cd $TEST_HOME/local/runtime-config && \
-              ./setup_environment.sh && ./setup_environment.sh >> $BASH_ENV            
+              cd $TEST_HOME/runtime-config && \
+              ./setup_environment.sh && ./setup_environment.sh >> $BASH_ENV
+      - run:
+          name: Check build whether custom backend is needed
+          command: |
+            echo "$CUSTOM_BACKEND" > /tmp/repo/custom_backend_switch
+      - restore_cache:
+          # when custom_backend_switch file has '1' a custom backend
+          # is used and the more extended set of maven repo's will be attached
+          keys:
+            - maven-repo-v1-{{ checksum "custom_backend_switch" }}
+      - run:
+          name: Build custom backend
+          command: |
+              (grep -q 0 /tmp/repo/custom_backend_switch && echo "Build of custom backend is not needed.") || $TEST_HOME/docker_compose/build.sh
+          no_output_timeout: 25m
```

Comment:
@pvannierop the `build.sh` prolly doesn't work b/c it would need maven right? Or is maven included in the ubuntu image